x/gov/keeper/proposal_test.go

```diff
@@ -192,38 +193,80 @@ func (suite *KeeperTestSuite) TestCancelProposal() {
 		suite.govKeeper.SetProposal(suite.ctx, proposal2)
 	}
 
+	proposal3, err := suite.govKeeper.SubmitProposal(suite.ctx, []sdk.Msg{prop}, "", "title", "summary", suite.addrs[2], false)
+	suite.Require().NoError(err)
+	proposal3ID := proposal3.Id
+
+	// add votes for proposal 3
+	suite.Require().NoError(suite.govKeeper.ActivateVotingPeriod(suite.ctx, proposal3))
+
+	proposal3, err = suite.govKeeper.Proposals.Get(suite.ctx, proposal3ID)
+	suite.Require().Nil(err)
+	suite.Require().True(proposal3.VotingStartTime.Equal(suite.ctx.BlockHeader().Time))
+	// add vote
+	voteOptions := []*v1.WeightedVoteOption{{Option: v1.OptionYes, Weight: "1.0"}}
+	err = suite.govKeeper.AddVote(suite.ctx, proposal3ID, suite.addrs[0], voteOptions, "")
+	suite.Require().NoError(err)
+
 	testCases := []struct {
 		name        string
+		malleate    func()
 		proposalID  uint64
 		proposer    string
 		expectedErr bool
 	}{
 		{
 			name:        "without proposer",
+			malleate:    func() {},
 			proposalID:  1,
 			proposer:    "",
 			expectedErr: true,
 		},
 		{
 			name:        "invalid proposal id",
+			malleate:    func() {},
 			proposalID:  1,
 			proposer:    string(suite.addrs[0]),
 			expectedErr: true,
 		},
 		{
 			name:        "valid proposalID but invalid proposer",
+			malleate:    func() {},
 			proposalID:  proposalID,
 			proposer:    suite.addrs[1].String(),
 			expectedErr: true,
 		},
 		{
 			name:        "valid proposalID but invalid proposal which has already passed",
+			malleate:    func() {},
 			proposalID:  proposal2ID,
 			proposer:    suite.addrs[1].String(),
 			expectedErr: true,
 		},
 		{
 			name:        "valid proposer and proposal id",
+			malleate:    func() {},
+			proposalID:  proposalID,
+			proposer:    suite.addrs[0].String(),
+			expectedErr: false,
+		},
+		{
+			name: "valid case with deletion of votes",
```

Comment:
I am not sure the test does what it says.

Reply:
Here the test doesn't delete votes, when `CancelProposal` is executed then the votes will be deleted.